test/k8s/service_helpers.go

```diff
@@ -217,12 +222,17 @@ func testCurlFromOutsideWithLocalPort(kubectl *helpers.Kubectl, ni *helpers.Node
 			"Can not connect to service %q from outside cluster (%d/%d)", url, i, count)
 		if checkSourceIP {
 			// Parse the IPs to avoid issues with 4-in-6 formats
-			sourceIP := net.ParseIP(strings.TrimSpace(strings.Split(res.Stdout(), "=")[1]))
-			var outIP net.IP
-			if sourceIP.To4() != nil {
-				outIP = net.ParseIP(ni.OutsideIP)
-			} else {
-				outIP = net.ParseIP(ni.OutsideIPv6)
+			ipStr := strings.TrimSpace(strings.Split(res.Stdout(), "=")[1])
+			sourceIP, err := netip.ParseAddr(ipStr)
+			ExpectWithOffset(1, err).Should(BeNil(), "Cannot parse IP %q", ipStr)
+			var outIP netip.Addr
+			if sourceIP.Is4() {
+				outIP, err = netip.ParseAddr(ni.OutsideIP)
+				ExpectWithOffset(1, err).Should(BeNil(), "Cannot parse IPv4 address %q", ni.OutsideIP)
+			}
+			if sourceIP.Is6() && !sourceIP.Is4In6() {
```

Comment:
If `sourceIP.Is4In6()` is true, then `outIP` is not set and it will not be equal to `sourceIP` which causes tests failures that we see now.

I think we need to treat `sourceIP.Is4In6()` being true as a separate case that needs to be handled in this test.